### PR TITLE
MAINT: Update Python version support documentation

### PR DESCRIPTION
**Pull request by j-t-1**

### CODE REVIEW

**Comment by j-t-1** [comment]
Do pypdf 3.x or PyPDF2 >= 2.0 support Python 3.12? If so, I will add a column to the table.

**Comment by pubpub-zz** [comment]
> Do pypdf 3.x or PyPDF2 >= 2.0 support Python 3.12? If so, I will add a column to the table.

PyPDF2 is now considered as obsolete. We will not support it anymore. pypdf  3.x, was not tested against Python 3.12 even if it could work. I do not recommend providing any hints about it.

File: docs/user/installation.md [hunk 1]
```diff
@@ -50,11 +50,17 @@ pip install pypdf[image]
 
 ## Python Version Support
 
+Since pypdf 4.0, every release, including point releases, should work with all
+supported versions of [Python](https://devguide.python.org/versions/). Thus
+every point release is designed to work with all concurrent Python versions,
```
Comment (MartinThoma) [inline docs/user/installation.md:55]:
```suggestion
every point release is designed to work with the current Python versions,
```

**Review by MartinThoma — DISMISSED** [review]
Looks good to me :+1: Thanks for the improvement :-)

**Review by stefan6419846 — APPROVED** [review]

**Pull request merged**